mm2src/coins/z_coin.rs

```diff
@@ -734,6 +740,47 @@ pub async fn z_coin_from_conf_and_params(
     builder.build().await
 }
 
+/// open file and calculate its sha256 digest as lowercase hex string
+fn sha256_digest(path: &PathBuf) -> Result<String, ZCoinBuildError> {
+    let input = File::open(path)?;
+    let mut reader = BufReader::new(input);
+
+    let digest = {
+        let mut hasher = Sha256::new();
+        let mut buffer = [0; 1024];
+        loop {
+            let count = reader.read(&mut buffer)?;
+            if count == 0 {
+                break;
+            }
+            hasher.update(&buffer[..count]);
+        }
+        format!("{:x}", hasher.finalize())
+    };
+    Ok(digest)
+}
+
+fn verify_checksum_zcash_params(spend_path: &PathBuf, output_path: &PathBuf) -> Result<bool, ZCoinBuildError> {
+    let spend_hash = sha256_digest(spend_path)?;
+    let out_hash = sha256_digest(output_path)?;
+    Ok(spend_hash == SAPLING_SPEND_EXPECTED_HASH && out_hash == SAPLING_OUTPUT_EXPECTED_HASH)
+}
+
+fn get_spend_output_paths(params_dir: PathBuf) -> Result<(PathBuf, PathBuf), ZCoinBuildError> {
+    let (spend_path, output_path) = if params_dir.exists() {
+        (
+            params_dir.join(SAPLING_SPEND_NAME),
+            params_dir.join(SAPLING_OUTPUT_NAME),
+        )
+    } else {
+        return Err(ZCoinBuildError::ZCashParamsNotFound);
+    };
+    if !(spend_path.exists() && output_path.exists()) {
+        return Err(ZCoinBuildError::ZCashParamsNotFound);
+    }
```

Comment:
In my opinion, the following code looks more readable because it includes less blocks.
```rs
if !params_dir.exists() {
    return Err(ZCoinBuildError::ZCashParamsNotFound);
}

let spend_path = params_dir.join(SAPLING_SPEND_NAME);
let output_path = params_dir.join(SAPLING_OUTPUT_NAME);

if !(spend_path.exists() && output_path.exists()) {
    return Err(ZCoinBuildError::ZCashParamsNotFound);
}
```

Just sharing my thought, maybe it's just me :)

Reply:
I agree, it looks better.
Done